src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -0,0 +1,1218 @@
+     * @return {@code true} if the corresponding variant is in a coding region; {@code false} otherwise.
+     */
+    @VisibleForTesting
+    static boolean isVariantInCodingRegion(final GencodeFuncotation.VariantClassification varClass,
+                                           final GencodeFuncotation.VariantClassification secondaryVarClass ) {
+
+        Utils.nonNull( varClass );
+
+        if ( varClass == GencodeFuncotation.VariantClassification.SPLICE_SITE ) {
+            Utils.nonNull( secondaryVarClass );
+            return secondaryVarClass != GencodeFuncotation.VariantClassification.INTRON;
+        }
+        else {
+            return  (varClass == GencodeFuncotation.VariantClassification.MISSENSE)        ||
+                    (varClass == GencodeFuncotation.VariantClassification.NONSENSE)        ||
+                    (varClass == GencodeFuncotation.VariantClassification.NONSTOP)         ||
+                    (varClass == GencodeFuncotation.VariantClassification.SILENT)          ||
+                    (varClass == GencodeFuncotation.VariantClassification.IN_FRAME_DEL)    ||
+                    (varClass == GencodeFuncotation.VariantClassification.IN_FRAME_INS)    ||
+                    (varClass == GencodeFuncotation.VariantClassification.FRAME_SHIFT_INS) ||
+                    (varClass == GencodeFuncotation.VariantClassification.FRAME_SHIFT_DEL) ||
+                    (varClass == GencodeFuncotation.VariantClassification.START_CODON_SNP) ||
+                    (varClass == GencodeFuncotation.VariantClassification.START_CODON_INS) ||
+                    (varClass == GencodeFuncotation.VariantClassification.START_CODON_DEL);
+        }
+    }
+
+    /**
+     * Creates a {@link List} of {@link GencodeFuncotation}s based on the given {@link VariantContext}, {@link Allele}, and {@link GencodeGtfGeneFeature}.
+     * @param variant The variant to annotate.
+     * @param altAllele The allele of the given variant to annotate.
+     * @param gtfFeature The GTF feature on which to base annotations.
+     * @return A {@link List} of {@link GencodeFuncotation}s for the given variant, allele and gtf feature.
+     */
+    @VisibleForTesting
+    List<GencodeFuncotation> createFuncotations(final VariantContext variant, final Allele altAllele, final GencodeGtfGeneFeature gtfFeature, final ReferenceContext reference) {
+        // For each applicable transcript, create an annotation.
+
+        final List<GencodeFuncotation> gencodeFuncotations = new ArrayList<>();
+
+        // TODO: instead of getting the best one here, we do them all, then in the renderer we condense them into 1 annotation based on worst effect.
+        // Get our "best" transcript:
+        final int bestTranscriptIndex = getBestTranscriptIndex(gtfFeature, variant);
+        if ( bestTranscriptIndex == -1 ) {
```

Comment:
Would this properly render IGR?

Use more specific GATKException, if possible.

Reply:
No - IGRs are handled in their own method `createIgrFuncotations`

Reply:
I'm not sure what other kind of `GATKException` would be better.

Reply:
Can this ever happen in practice?  If not, ShouldNotGetHere.  

If so, is this a user error?